com.github.sdbg.debug.core/src/com/github/sdbg/debug/core/internal/util/DebuggerUtils.java

```diff
@@ -80,39 +80,7 @@ public static String demangleVmName(String name) {
       return null;
     }
 
-    int atIndex = name.indexOf('@');
```

Comment:
Putting the GWT SuperDevMode name demangling here can be done as a (very) short term solution.
DebuggerUtils and especially method demangleVmName() are DART-isms which should be removed from SDBG, as we are not dealing with the DART VM at all.

Further, unconditionally demangling all variable names could be a bit dangerous. What if this is not a GWT variable? 
* E.g. you are debugging a 3rd party library written in regular JavaScript?
* Or you are debugging, say, a CoffeeScript JS output, where CoffeeScript has a completely different variable naming rules?

I think we have to meet the following objectives for variables / method names' demangling in GWT:
* It has to be possible to switch it on or off. Also initially, it has to probably be in off state
* Ideally, It has to be done only for GWT variables. Ideally, only for GWT SuperDevMode variables. Plain JavaScript variables or other code variables (CoffeeScript etc.) should be unaffected

At least for variables, a promising (IMO) way we can explore is to use the "Logical Structure" notion of Eclipse:
* It allows you to replace not just the name of the variable, but the whole variable structure with something custom. 
* It can be switched on or off with a toggle in the Variables view
* It is contributed in a nice clean way by implementing ILogicalStructureTypeDelegate(2) or ILogicalStructureProvider. Even the Dart code we've inherited does that - take a look at MapStructureType and how it is contributed in the plugin.xml (yes, MapStructureType should be removed as well)
* This approach may work very well for the generic solution @skybrian & ChromeDevTools are developing as well (check  issue  #6)

Please let me know your thoughts, thanks.